pkg/gen/nodejs-templates/yaml.ts.mustache

```diff
@@ -120,81 +120,6 @@ import * as outputs from "../types/output";
         resourcePrefix?: string;
     }
 
-    function yamlLoadAll(text: string): any[] {
-        // NOTE[pulumi-kubernetes#501]: Use `loadAll` with `JSON_SCHEMA` here instead of
-        // `safeLoadAll` because the latter is incompatible with `JSON_SCHEMA`. It is
-        // important to use `JSON_SCHEMA` here because the fields of the Kubernetes core
-        // API types are all tagged with `json:`, and they don't deal very well with things
-        // like dates.
-        const jsyaml = require("js-yaml");
-        return jsyaml.loadAll(text, undefined, {schema: jsyaml.JSON_SCHEMA});
-    }
-
-    /** @ignore */ export function parse(
```

Comment:
Have we actually save much/anything by moving just `parse` into Go?  It seems most of the code in the implementation of `parse` here ends up getting duplicated back into callers anyway.  What pieces of code are we actually moving?  As close as I can tell - we've only really moved the implementation of `yamlLoadAll`, but instead of just moving that (which would have been much simpler I believe), we've put the boundary somewhere else which leads to a more complex refactoring of a lot of other code?

I can't quite follow - but it seems to first approximation as though we're doing either too much or too little - and either of these would be better:
1. Move just `yamlLoadAll` into Go
2. Move `helm template` and as much code as possible dependent on the result of it into Go

Very possible I'm missing something about why the approach taken here was a natural place to draw a line.

Reply:
My plan was to move ~everything except the transformations and resource creation into the provider, but I got sidetracked on the performance problems I was seeing, and wanted to understand that better before continuing.